### PR TITLE
feat(sdk): adds support for priority_class_name. Fixes #3594

### LINKED ISSUE #3594: [Question] can we set priorityClassName or priority to pods in kubeflow pipeline?
It's a pure question rather than a bug.
Is there any trick to set priorityClassName or priority to pods in pipeline?

The argo workflow seems to support it now:
https://github.com/argoproj/argo/pull/1179/files

Here is some comments that kubeflow pipeline doesn't support it because it is "Not supported by Argo".
https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/components/structures/kubernetes/v1.py#L408

Thanks!

### PR DESCRIPTION
**Pull request by munagekar**
**Description of your changes:**
- Adds Support for Priority Class Name
- Fixes https://github.com/kubeflow/pipelines/issues/3594

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by munagekar** [comment]
Tested with Argo 2.3.0 works fine.
Ref : https://argoproj.github.io/argo/fields/#template

**Comment by munagekar** [comment]
/test kubeflow-pipeline-frontend-test
 Rerunning frontend-test since I can't see any logs and change shouldn't affect frontend.

**Comment by munagekar** [comment]
@chensun Test have succeeded. PTAL.

File: sdk/python/kfp/dsl/_container_op.py [hunk 1]
```diff
@@ -878,6 +879,16 @@ def add_pod_label(self, name: str, value: str):
         self.pod_labels[name] = value
         return self
 
+
+    def set_priority_class_name(self, name):
```
Comment (chensun) [inline sdk/python/kfp/dsl/_container_op.py:883]: nit: type hint: `name: str`

**Review by chensun — COMMENTED** [review]
/lgtm

Pass to @Ark-kun for final approval.

**Comment by chensun** [comment]
/lgtm 

Thanks!

**Comment by munagekar** [comment]
@Ark-kun Friendly Ping

**Comment by munagekar** [comment]
@Ark-kun Could you take a look at this PR?

**Comment by munagekar** [comment]
@numerology @Ark-kun Could you take a look at this PR ?

File: sdk/python/kfp/dsl/_container_op.py [hunk 2]
```diff
@@ -878,7 +879,16 @@ def add_pod_label(self, name: str, value: str):
         self.pod_labels[name] = value
         return self
 
-    def set_retry(self, num_retries: int, policy: str = None):
+    def set_priority_class_name(self, name: str):
+        """Adds a priority class to the pod
```
Comment (chensun) [inline sdk/python/kfp/dsl/_container_op.py:883]: Hi @munagekar , I recently formatted this file to use 2 spaces indentations. Can you please rebase your PR and make the indentation consistent within the file? Sorry about the hassle.

Once done rebasing, the PR can be merged.

**Comment by munagekar** [comment]
Accidentally closed the pull request and incorrectly force pushed.

File: sdk/python/kfp/dsl/_container_op.py (hunk 2 above)
Comment (munagekar) [reply sdk/python/kfp/dsl/_container_op.py]: @chensun I have rebased (kind of 😅 ). PTAL.

**Comment by Ark-kun** [comment]
Hello @munagekar 
Sorry for the review delay.
Could you please change the name to `pod_priority_class_name` and `set_pod_priority_class_name` to clearly indicate that this is a pod-level setting (ContainerOp has it's own properties plus container properties plus pod properties).
Apart from this, I think we can merge this PR.

**Comment by munagekar** [comment]
@Ark-kun PTAL

**Pull request closed without merging**